src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -853,43 +853,52 @@ def generate_flow_tools_json(
     return flow_tools
 
 
-def update_user_agent_from_env_var():
-    """Update user agent from env var to OperationContext"""
-    from promptflow._core.operation_context import OperationContext
+class ClientUserAgentUtil:
+    """SDK/CLI side user agent utilities."""
+
+    @classmethod
+    def _get_context(cls):
+        from promptflow._core.operation_context import OperationContext
 
-    if USER_AGENT in os.environ:
-        # Append vscode or other user agent from env
-        OperationContext.get_instance().append_user_agent(os.environ[USER_AGENT])
+        return OperationContext.get_instance()
 
-    if PF_USER_AGENT in os.environ:
-        # Append promptflow user agent from env
-        OperationContext.get_instance().append_user_agent(os.environ[PF_USER_AGENT])
+    @classmethod
+    def get_user_agent(cls):
+        context = cls._get_context()
+        # directly get from context since client side won't need promptflow/xxx.
+        return context.get(context.USER_AGENT_KEY, "").strip()
```

Comment:
Note that context.USER_AGENT_KEY is accessed using the instance (context) rather than the class (OperationContext). If USER_AGENT_KEY is a class variable of OperationContext, it would be more clear to access it using the class name.
```
    @classmethod  
    def get_user_agent(cls):  
        context = cls._get_context()  
        # Directly get from context since client side won't need promptflow/xxx.  
        return context.get(OperationContext.USER_AGENT_KEY, "").strip() 
```

Reply:
Updated